app/javascript/src/controller_pages.js

```diff
@@ -407,8 +407,18 @@ function addQuestionMenuListeners(view) {
       activator: question.menu.selectedItem,
       remote: true,
       autoOpen: true,
-      submitOnClickSelector: 'input[type="submit"]',
-      onSuccess: function(data) {
+      submitOnClickSelector: 'button[type="submit"]',
+      beforeSubmit: function(dialog) {
+        var $button = dialog.$submitButton; 
+        $button.text('Uploading...');
+        $button.attr('disabled', 'disabled');
+      },
+      onSuccess: function(_, dialog) {
+        var $success = question.$node.find('[data-fb-status="autocomplete-success"]');
+        var $warning =  question.$node.find('[data-fb-status="autocomplete-warning"]');
+        var $successMessage = question.$node.find('[data-fb-template="autocomplete-success-message"]').first().html();
+        $success.html($successMessage);
+        $warning.remove();
```

Comment:
I agree with the comments around `beforeSubmit` this is updating things within the dialog - the page controller shouldn't be doing that.  Not sure about your suggestion to fix - I think I have a more generic solution that would work.

For `onSuccess`, I think this should be in the controller it is responsible for updating things in the view on the page which is what we use the controller for elsewhere, and it would be strange for the dialog to be responsible for this.

The success message doesn't come from the server because it is an aria-live element, and also the success message needs to be in the page by default, and also if the page already has an upload the success message is displayed, which means the message is always in the page, and we are just controlling visibility (or injecting it into an aria-live region so that it is announced by screenreaders)